internal/controllers/device_config_controller.go

```diff
@@ -294,34 +252,29 @@ func (d *DeviceConfigController) GetConfigURLs(c *fiber.Ctx) error {
 	vin := c.Params("vin")
 
 	// Set up gRPC call to retrieve UserDevice by VIN
-	req := &pb.GetUserDeviceByVINRequest{Vin: vin}
-	ud, err := d.usersClient.GetUserDeviceByVIN(context.Background(), req)
+	ud, err := d.userDeviceSvc.GetUserDeviceByVIN(context.Background(), vin)
```

Comment:
instead of `context.Background()` you want to use the context that comes from the fiber request `c.Context`